shopfloor/services/checkout.py

```diff
@@ -913,11 +922,37 @@ def scan_package_action(self, picking_id, selected_line_ids, barcode):
 
         selected_lines = self.env["stock.move.line"].browse(selected_line_ids).exists()
         search_result = self._scan_package_find(picking, barcode)
+
+        message = self._check_scan_package_find_result(search_result)
+        if message:
+            return self._response_for_select_package(
+                picking,
+                selected_lines,
+                message=message,
+            )
+
         result_handler = getattr(
             self, "_scan_package_action_from_" + search_result.type
         )
         return result_handler(picking, selected_lines, search_result.record)
 
+    def _check_scan_package_find_result(self, search_result):
+        ppt = self.work.menu.package_process_type
+        # Currently there is no known way to finish the checkout with a scan. To
+        # process without any package is only possible with a button. In its def
+        # a BadRequest is raised in case no package is not allowed
+        if not ppt or ppt == "with_package":
+            return
+
+        stype = search_result.type
+
+        if ppt == "without_package" and stype in [
+            "package",
+            "packaging",
+            "delivery_packaging",
+        ]:
+            return self.msg_store.invalid_scanned_checkout_object_wo_package(stype, ppt)
+
```

Comment:
Could you please add a docstring and rename this method. 

```suggestion
    def _is_package_scan_allowed(self, search_result):
        if self._is_process_with_package_enabled():
            return
        package_process_type = self.work.menu.package_process_type
        if  package_process_type == "without_package" and search_result.type in [
            "package",
            "packaging",
            "delivery_packaging",
        ]:
            return self.msg_store.checkout_package_scan_is_not_allowed(search_result.type, package_process_type)

```